### ISSUE TITLE
[v15.0 Alpha] Rock Cleanup Exception in Update Person Age And Range

### ISSUE THREAD

**Issue by JimMichael** [SparkDevNetwork/Rock#5380]
### Please go through all the tasks below

- [X] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.

Seeing an exception in the Rock Cleanup job on v15.0 alpha, specifically in the Update Person Age and Range step.

### Expected Behavior

Rock Cleanup job would complete without error

### Actual Behavior

We get the following exception. I saw it when the job ran automatically the first time after upgrading, and when subsequently running the job manually.

![image](https://user-images.githubusercontent.com/3903408/230666657-5c3ec1a6-87f6-47d6-876b-89959652c34d.png)

### Steps to Reproduce

1. Go to Admin | System Settings | Jobs Administration

2. Run the Rock Cleanup job. 

I have not been able to determine if this happens on any v15 install or only our our db (which is a copy of production.)

### Rock Version

v15.0 aloph

### Client Culture Setting

En-us

**Comment by mikedotmundy** [comment]
Our dev instance on v1.15.0.16 is also throwing the same exception. Ours has the same Exception Type/Stack Trace's PLUS an additional stack trace not in the picture @JimMichael shared.

<img width="1571" alt="Screenshot 2023-04-08 at 09 37 07" src="https://user-images.githubusercontent.com/23226974/230724216-aece75cf-1907-4637-bcf9-f26506f59b92.png">

**Comment by ethan-sparkdevnetwork** [comment]
Thanks, I have added this to the Alpha results list.

**Comment by nairdo** [comment]
This one should be fixed by this commit that was just picked into the v15.0 branch: https://github.com/SparkDevNetwork/Rock/commit/1abb3406be0c19075258c12ed98873c16ae29145  (On the #alpha-testers channel, I'll post the updated Rock.dll for any alpha testers that want to re-test this too.)

**Comment by mikedotmundy** [comment]
Upon updating to v15.1, this exception hit again. However, it only happened the _first_ time this job ran. The 2nd time it ran (I manually ran it), it did complete successfully.

Here is the exception:
![Screenshot 2023-06-06 at 16 19 59](https://github.com/SparkDevNetwork/Rock/assets/23226974/25acfb0b-2e32-4cd3-874f-983fce9c6716)

Here are the 2 stack traces:
```
   at Rock.Jobs.RockCleanup.Execute() in \Rock\Jobs\RockCleanup.cs:line 324
   at Rock.Jobs.RockJob.Quartz.IJob.Execute(IJobExecutionContext context) in \Rock\Jobs\RockJob.cs:line 142
   at Quartz.Core.JobRunShell.Run()
```

```
 at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at System.Data.Entity.Infrastructure.Interception.InternalDispatcher`1.Dispatch[TTarget,TInterceptionContext,TResult](TTarget target, Func`3 operation, TInterceptionContext interceptionContext, Action`3 executing, Action`3 executed)
   at System.Data.Entity.Infrastructure.Interception.DbCommandDispatcher.NonQuery(DbCommand command, DbCommandInterceptionContext interceptionContext)
   at System.Data.Entity.Core.Objects.ObjectContext.ExecuteStoreCommandInternal(String commandText, Object[] parameters)
   at System.Data.Entity.Core.Objects.ObjectContext.ExecuteInTransaction[T](Func`1 func, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction, Boolean releaseConnectionOnSuccess)
   at System.Data.Entity.Core.Objects.ObjectContext.<>c__DisplayClass174_0.b__0()
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
   at Rock.Jobs.RockCleanup.CalculateAgeAndAgeBracketOnAnalyticsSourceDate() in \Rock\Jobs\RockCleanup.cs:line 2785
   at Rock.Jobs.RockCleanup.UpdateAgeAndAgeBracketOnPerson() in \Rock\Jobs\RockCleanup.cs:line 2825
   at Rock.Jobs.RockCleanup.RunCleanupTask(String cleanupTitle, Func`1 cleanupMethod) in \Rock\Jobs\RockCleanup.cs:line 437

The wait operation timed out
```

**Comment by mikedotmundy** [comment]
I am running v15.1 in production now and am still seeing the exception in the comment above about every other day when the job runs.

**Comment by dataCollegechurch** [comment]
@mikedotmundy - do you by chance scale your database down overnight when the job runs. My job was also failing when it was automatically overrun with lower resources.

**Comment by dataCollegechurch** [comment]
@nairdo - should @mikedotmundy and I open a new issue or keep commenting here. 

I think we many need to optimize [the associated UPDATE Statement here](https://github.com/SparkDevNetwork/Rock/blob/e0aa9d6d737bd1e9ad473339f4536146eaa8956a/Rock/Jobs/RockCleanup.cs#L2787)

AnalyticsSourceDate is going to include 55392+ records regardless of the Rock instance in question and that is a lot of records to update. Perhaps we could be more targeted and only update the records that need to be.

**Comment by mikedotmundy** [comment]
No, I do not scale mine at all.

**Comment by nairdo** [comment]
@mikedotmundy Mike and @dataCollegechurch Tony, I'm looking at the history of code changes to that job and I see there was a fix made last month (https://github.com/SparkDevNetwork/Rock/commit/631dcee3ccf39e6e55bbc94e7cf35786f9340b3d) in the v15.2 branch -- and that change now uses the job's "**Command Timeout**" setting:

was `using ( var rockContext = RockContext() )` but is now:
[`using ( var rockContext = CreateRockContext() )`](https://github.com/SparkDevNetwork/Rock/commit/631dcee3ccf39e6e55bbc94e7cf35786f9340b3d#diff-5b283f91e011def577f556bddf3a0fb7e2ac222b02bc47da6c8b289961bfd596R2725)

Previously the operation would have had to complete in 30 seconds or less, but now it would not timeout for 15 minutes (that should be an excessive amount of time needed to complete that SQL).  So, based on that change, I'm going to update the **Fixed in** label to **v15.2**.  If you're an an alpha tester, you should also be able to confirm the fix is in v16.0 (since the v15.2 changes are also in v16.0).  Otherwise please let us know by reopening this issue once you get either v15.2 or v16.0.

**Comment by mikedotmundy** [comment]
Thank you @nairdo !
